dfget/core/downloader/p2p_downloader/client_writer.go

```diff
@@ -220,6 +221,7 @@ func (cw *ClientWriter) write(piece *Piece) error {
 	}
 
 	if cw.acrossWrite {
+		atomic.AddInt32(&piece.writerNum, 1)
```

Comment:
It's better to abstract this line as a method of struct `Piece`, such as `Piece.IncWriter`.